homeassistant/components/keba/__init__.py

```diff
@@ -29,6 +30,7 @@
         DOMAIN: vol.Schema(
             {
                 vol.Required(CONF_HOST): cv.string,
+                vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
```

Comment:
Is this really necessary? Both the name and the entity_id of the entities can be changed by the user in the GUI since we provide a unique id for the entities. If more than one charging station could be configured, this option would be more reasonable. But at the moment only one station can be configured.

Reply:
Hi Martin,
thanks for your comment!
In the current version, e.g. the sensor that holds the total energy consumed by the charging station is called "sensor.total_energy". From this default entity id, it is hard to understand what exactly it should represent as it could also be the total energy of my smart meter. After figuring out to which device this sensor belongs, it can be renamed in the frontend of course. This change to have a better discoverability of the sensors was requested by other users in the forum.

Do you think, it is better to use the product name instead, the charging station is sending me in the initial request? That would be like the charging station that we are using for the unique_id?

Reply:
If the default entity_id is hard to identify, we should update the default entity_id to be easier to identify. We don't need to provide a name option for that.

Reply:
In my last PR it has been justly criticised to add `keba_` in front of all entity_ids as that would have been hard coded (that was my understanding).
One solution I am thinking about is the following. From the charging station, I receive this json:
```
{
"ID": "1",                                 // requested report ID
"Product": "BMW-10-EC240522-E1R",          // product name
"Serial": "18104361",                      // serial used for unique_id
"Firmware":"P30 v 3.08.4 (170307-132509)", // firmware verison
"COM-module": 0,                           // status of an optional extension card
"Sec": 30132129                            // total seconds uptime
}
```
As the product name is also not very readable, it is classified in the device library to `KEBA P30`, `KEBA P20` or `BMW Wallbox`. How about putting this classified product name in front of the entity_id like this `sensor.bmw_wallbox_total_energy`?

Reply:
If you're referring to this comment https://github.com/home-assistant/home-assistant/pull/24484#discussion_r305488323, it was aimed at the `unique_id` attribute, not the `entity_id`.

You know the device best. If you think the users will identity the device by adding the product name to the entity_id, that sounds good. :+1:

Reply:
ah ok, thanks for clarifying! It seems I mixed something up.